### ISSUE TITLE
EnumMember value not used when serializing

### ISSUE THREAD

**Issue by dnasir** [reactiveui/refit#459]
The library works great when I'm retrieving data from my REST API, and so far it manages to correctly map all of my enums. However, when I need to use the same enums in my requests as querystring parameters, Refit doesn't seem to use the `EnumMember` value set in my data contracts.

Example:
```csharp
[DataContract]
[JsonConverter(typeof(StringEnumConverter))]
public enum ServiceType
{
    [EnumMember(Value = "ondemand")]
    Ondemand = 1,

    [EnumMember(Value = "flex")]
    Flex = 2,

    [EnumMember(Value = "carpool")]
    Carpool = 3
}
```

When I debug this, the request sent out would use the enum property names rather than the `EnumMember` values. i.e. `https://api-endpoint/services?type=Flex` instead of `https://api-endpoint/services?type=flex`. This causes the endpoint to return an exception.

I managed to solve this by implementing my own `IUrlParameterFormatter`, like so:

```csharp
public class EnumAsEnumMemberValueParameterFormatter : IUrlParameterFormatter
{
    public string Format(object parameterValue, ParameterInfo parameterInfo)
    {
        if(parameterValue == null)
            return null;

        var type = parameterValue.GetType();
        if(!type.IsEnum)
            return parameterValue.ToString();

        var enumValue = type.GetMember(parameterValue.ToString());
        if(enumValue.Length == 0)
            return null;

        var attr = enumValue[0].GetCustomAttribute<EnumMemberAttribute>();
        return attr != null ? attr.Value : enumValue.ToString();
    }
}
```

But I feel like this is something that should be taken care of by either Refit or the Json.NET library. Maybe I missed something.

**Comment by dnasir** [comment]
I found that the following code works just as well:

```csharp
public class JsonConvertParameterFormatter : IUrlParameterFormatter
{
    public string Format(object parameterValue, ParameterInfo parameterInfo)
    {
        return parameterValue == null ? null : JsonConvert.SerializeObject(parameterValue).Replace("\"", "");
    }
}
```

I'm not sure why `JsonConvert.SerializeObject()` would add double quotes around the output, and I'm not that well-versed in Json.NET to figure it out. Maybe someone here can give me some pointers regarding this.

**Comment by bennor** [comment]
@dnasir We don't have support for the `EnumMemberAttribute` yet. 

The reason `JsonConvert.SerializeObject(parameterValue)` gives you a quoted string is that you've converted a string to a JSON representation of a string -- which is always enclosed in quotes.

**Comment by Styxxy** [comment]
@dnasir This should have been fixed now with PR #502 .

**Comment by clairernovotny** [comment]
Closing due to age. Please try Refit v6 and reopen if still an issue.
